49.md

```diff
@@ -0,0 +1,83 @@
+NIP-49
+======
+
+Nostr Wallet Auth
+--------------------
+
+`draft` `optional` `author:benthecarman`
+
+## Rationale
+
+[Nostr Wallet Connect](47.md) is a protocol for connecting a Nostr client to a remote Lightning wallet. It is great but
+hinges on having the user copy-paste a wallet connection URI into their client. This can be a UX hurdle and often has
+the user handling sensitive information that they may not understand.
+
+As well, Nostr Wallet Connect has lots of different functions that a wallet may or may not support or a service may or
+may not need. This either runs into silent failures or requires the user to understand what functions their wallet
+supports and what functions the service needs.
+
+Finally, Nostr Wallet Connect requires the wallet service to generate both key pairs and the service to use these. This
+limits the key management strategies that a service can use and often leads to suboptimal key management.
+
+This NIP proposes a new protocol that solves these problems by having the client and service generate a NWC connection
+URI together. This URI is then used to connect the client and service.
+
+## Theory of Operation
+
+1. The user wants to connect their wallet to a service. The service provides a QR code or other means of connecting to
+   the service. The QR code contains a `nip49://` URI with a `service_key` parameter that contains the service's public
+   key, the relay to communicate over, and a list of functions that the service requires.
+2. The user scans the QR code and their wallet prompts them the required functions and asks them to confirm that they
+   want to connect to the service.
+3. On confirmation, the wallet generates a new key pair and sends the public key to the service in a confirmation event.
+4. With the combination of both keys, the wallet and the service both will generate the same connection URI. They can
+   now use NWC to communicate.
+
+## Specification
+
+### URI Format
+
+The **service** generates this connection URI with protocol `nip49:` and base path it's hex-encoded `pubkey` with the
+following query string parameters:
+
+- `relay` Required. URL of the relay where the **wallet service** is connected and will be listening for events. May be
+  more than one.
+- `required_commands` Required. A space-separated list of commands that the **wallet service** requires. The **client**
+  MUST NOT
+  connect if it does not support all of these permissions.
+- `optional_commands` Optional. A space-separated list of commands that the **wallet service** can enable to add
+  additional functionality. The **client** MAY ignore these.
+- `budget` Optional. A budget that the **wallet service** will use to limit the amount of funds that the **service** can
```

Comment:
Is this budget related to a specific account the service manages or is it a max budget for all accounts? The account doesn't seem to be included in the URI, so I am confused. I am not sure if the client would care about a budget of all accounts.

Reply:
this is the budget for the single connection, it is in the uri